packages/gitbook/src/app/middleware/(site)/(core)/sitemap-pages.xml/route.ts

```diff
@@ -0,0 +1,100 @@
+import { RevisionPage, RevisionPageDocument, RevisionPageGroup } from '@gitbook/api';
+import jsontoxml from 'jsontoxml';
+
+import { getRevisionPages, getSpace } from '@/lib/api';
+import { getAbsoluteHref, getBasePath } from '@/lib/links';
+import { getPagePath } from '@/lib/pages';
+import { getSiteContentPointer } from '@/lib/pointer';
+import { isPageIndexable } from '@/lib/seo';
+
+export const runtime = 'edge';
+
+/**
+ * Generate a sitemap.xml for the current space.
+ */
+export async function GET() {
+    const pointer = await getSiteContentPointer();
+
+    const revisionId = await (async () => {
+        if (pointer.revisionId) {
+            return pointer.revisionId;
+        }
+        const space = await getSpace(pointer.spaceId, pointer.siteShareKey);
+        return space.revision;
+    })();
```

Comment:
I'm not sure it's worth optimizing for this; the only case that matters in reality will be the case where `pointer.revisionId` is not defined

```suggestion
    const revisionId = pointer.revisionId ?? (await getSpace(pointer.spaceId, pointer.siteShareKey).revision);
```